sdk/storage/azure-storage-files-shares/src/share_lease_client.cpp

```diff
@@ -16,10 +17,12 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
 
   Azure::Response<Models::AcquireLeaseResult> ShareLeaseClient::Acquire(
       std::chrono::seconds duration,
-      const AcquireLeaseOptions& options,
+      const AcquireLeaseOptions&,
       const Azure::Core::Context& context) const
   {
-    (void)options;
+
+    AZURE_ASSERT(m_fileClient.HasValue() || m_shareClient.HasValue());
```

Comment:
We don't need this

Reply:
I think we do.
Check below if...elseif...  the last else{} is aborting, which means this method has a pre-condition about either m_file or m_shareClient needs to has a value.  Whenever that's not true, there's a bug in the app that needs to be fixed and can't be recovered.

I would ask why are we not throwing and invalid argument exception here instead, but that's storage choice and I am just transforming the current abort() approach to asserting things are like expected to work

Reply:
> Check below if...elseif... the last else{} is aborting, which means this method has a pre-condition about either m_file or m_shareClient needs to has a value. Whenever that's not true, there's a bug in the app that needs to be fixed and can't be recovered.

If both of them are empty, it will crash in the `else {}` block.

Reply:
This is an ugly implementation. Maybe consider to enhance it.  Or maybe it is because it is the auto-gen, but still ugly.
Most of the code is repeated inside each if().  I would create a template function to remove the code duplication with the input for the client to be used and the types.

Then, this method should have a precondition about it needs at least one client to work (and maybe not both).  

But that's just my opinion LOL.  Coming back to the precondition, sure, I can remove it and let the ifs do the job

Reply:
Being ugly or not can be very subjective.

Let's just replace abort with ASSERT macro, to keep this PR single purpose, and leave alone code logic.